virttest/utils_test/utils_nwfilter.py

```diff
@@ -1,10 +1,79 @@
 """
+Virtualization test - network filter related utility functions
 Module simplifying manipulation of XML described at
 http://libvirt.org/formatnwfilter.html
 """
 
+import logging
+
 from virttest.libvirt_xml import base, accessors
 from virttest.libvirt_xml.devices.interface import Filterref
+from virttest import libvirt_xml
+
+# network filter xml set up code BEGIN
+
+
+def set_nwfilter_iface(New_iface,
+                       type_name="network",
+                       source={'network': "default"},
+                       filterref_dict={}):
+    """
+    set iface to bind network or bind filter
+
+    Params New_iface: instance of Interface,
+    Params type_name: the type name which bind to interface
+    Params source: source of new_iface, set as default
+    Params filterref_dict: dict of network filter, which is not
+    blank will bind to new_iface
+
+    return:  new network interface
+    """
+    # set iface type name as network and binding source type
+    New_iface.type_name = type_name
+    New_iface.source = source
+    # if filterref_dict is not blank , bind it to iface
+    if filterref_dict:
+        filterref = New_iface.new_filterref(**filterref_dict)
+        New_iface.filterref = filterref
```

Comment:
sometimes this line doesn't work,  you can try below code. The New_iface is got from existing vmxml and does not ever contain a filterref property. Then accessor check will be failed. 
```
         vmxml = vm_xml.VMXML.new_from_dumpxml(vm_name)
-        xml_devices = vmxml.devices
-        iface_index = xml_devices.index(
-            xml_devices.by_device_tag("interface")[0])
-        iface = xml_devices[iface_index]
-        iface.type_name = "bridge"
-        iface.source = {'bridge': br_name}
-        iface.filterref = iface.new_filterref(name=nwfilter)
-        iface.xmltreefile.write()
```
```
File "/var/tmp/libvirt-ci/runtest/avocado-vt/avocado-vt/virttest/libvirt_xml/accessors.py", line 29, in type_check
    % (name, expected, is_a_name))

ValueError: Instance of Filterref value is not any of [<class 'interface.Filterref'>], it is a <class 'interface.Filterref'>
```